appstore/notification.go

```diff
@@ -32,27 +35,41 @@ type NotificationExpiresDate struct {
 }
 
 type NotificationReceipt struct {
-	UniqueIdentifier          string `json:"unique_identifier"`
-	AppItemID                 string `json:"app_item_id"`
-	Quantity                  string `json:"quantity"`
-	VersionExternalIdentifier string `json:"version_external_identifier"`
-	UniqueVendorIdentifier    string `json:"unique_vendor_identifier"`
-	WebOrderLineItemID        string `json:"web_order_line_item_id"`
-	ItemID                    string `json:"item_id"`
-	ProductID                 string `json:"product_id"`
-	BID                       string `json:"bid"`
-	BVRS                      string `json:"bvrs"`
-	TransactionID             string `json:"transaction_id"`
-	OriginalTransactionID     string `json:"original_transaction_id"`
-	IsTrialPeriod             string `json:"is_trial_period"`
-	IsInIntroOfferPeriod      string `json:"is_in_intro_offer_period"`
+	UniqueIdentifier            string `json:"unique_identifier"`
+	AppItemID                   string `json:"app_item_id"`
+	Quantity                    string `json:"quantity"`
+	VersionExternalIdentifier   string `json:"version_external_identifier"`
+	UniqueVendorIdentifier      string `json:"unique_vendor_identifier"`
+	WebOrderLineItemID          string `json:"web_order_line_item_id"`
+	ItemID                      string `json:"item_id"`
+	ProductID                   string `json:"product_id"`
+	BID                         string `json:"bid"`
+	BVRS                        string `json:"bvrs"`
+	TransactionID               string `json:"transaction_id"`
+	OriginalTransactionID       string `json:"original_transaction_id"`
+	IsTrialPeriod               string `json:"is_trial_period"`
+	IsInIntroOfferPeriod        string `json:"is_in_intro_offer_period"`
+	IsUpgraded                  string `json:"is_upgraded"`
+	PromotionalOfferID          string `json:"promotional_offer_id"`
+	SubscriptionGroupIdentifier string `json:"subscription_group_identifier"`
 
 	PurchaseDate
 	OriginalPurchaseDate
 	NotificationExpiresDate
 	CancellationDate
 }
 
+type NotificationUnifiedReceipt struct {
+	Status string `json:"status"`
+
+	// For production we will get "Production" instead of "PROD"
+	Environment NotificationEnvironment `json:"environment"`
+
+	LatestReceipt      string                `json:"latest_receipt"`
+	LatestReceiptInfo  []NotificationReceipt `json:"latest_receipt_info"`
```

Comment:
Thanks for your PR :-)

[The document says](https://developer.apple.com/documentation/appstoreservernotifications/unified_receipt) that ``unified_receipt`` has [this receipt](https://developer.apple.com/documentation/appstorereceipts/responsebody/latest_receipt_info)(not [S2S notification's receipt](https://developer.apple.com/documentation/appstoreservernotifications/responsebody/latest_receipt_info)).

So you should use ``InApp`` struct instead of ``NotificationReceipt``.

Reply:
Ah, thank you!
This is fixed now.